test/e2e/e2e_suite_test.go

```diff
@@ -208,6 +214,38 @@ func initBootstrapCluster(bootstrapClusterProxy framework.ClusterProxy, config *
 	}, config.GetIntervals(bootstrapClusterProxy.GetName(), "wait-controllers")...)
 }
 
+func dumpBootstrapClusterLogs(bootstrapClusterProxy framework.ClusterProxy) {
+	if bootstrapClusterProxy == nil {
+		return
+	}
+
+	clusterLogCollector := bootstrapClusterProxy.GetClusterLogCollector()
+	if clusterLogCollector == nil {
+		return
+	}
+
+	nodes, err := bootstrapClusterProxy.GetClientSet().CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("Failed to get nodes for the bootstrap cluster: %v\n", err)
+		return
+	}
+
+	// We assume that the test framework generated bootstrap cluster has only one node.
+	bootstrapNodeName := nodes.Items[0].GetName()
```

Comment:
Given that we have already have the list of nodes, I think we should use it and loop through all the nodes.
This makes us more flexible in case users rely on existing Kubernetes clusters.

Reply:
Done!